src/Yesod/Paginator/Widgets.hs

```diff
@@ -178,10 +178,12 @@ getUpdateGetParams
 getUpdateGetParams pageParamName = do
     params <- handlerToWidget $ reqGetParams <$> getRequest
     pure
-        $ \number ->
-              nubOn fst
-                  $ [(unPageParamName pageParamName, tshow number)]
-                  <> params
+        $ \number -> formatParams number params
+  where
+    formatParams :: Show a => a -> [(Text, Text)] -> [(Text, Text)]
+    formatParams number params =
+      let name = unPageParamName pageParamName
+      in [(name, tshow number)] <> filter ((/=) name . fst) params
```

Comment:
I'm not sure why the `params` list contains multiple parameters with the same name as `pageParamName` and wether filtering those out has unintended consequences, but all tests seem to pass and it works fine for my single use case.

Reply:
Oh yeah, that was lazy. We we're unique-in by name since we were adding our `pageParamName` to a list that would already contain one. The `((/=) . name)` approach makes more sense.

Alternatively, you could also write some sort of `replaceBy` and use that. For example,

```hs
formatParams number =
  replaceBy ((==) `on` fst) (unPageParamName pageParamName, tshow number)

replaceBy
  :: (a -> a -> Bool)
  -> a
  -> [a]
  -> [a]
replaceBy f x [] = []
replaceBy f x (y:ys)
  | f x y = x : replaceBy f x ys
  | otherwise = y: replaceBy f x ys
```

Reply:
Nevermind about `replaceBy`. I now realize we need a replace-or-append sort of thing. Not easy to generalize.